src/learner.cc

```diff
@@ -648,6 +662,10 @@ class LearnerImpl : public Learner {
   std::vector<std::shared_ptr<DMatrix> > cache_;
 
   common::Monitor monitor_;
+
+  /*! \brief saved config keys used to restore failed worker */
+  std::set<std::string> saved_configs_ = {"max_depth", "tree_method", "dsplit",
+    "seed", "silent", "num_round", "gamma", "min_child_weight"};
```

Comment:
How many of these are new?

Reply:
all. those are missing configs to rerun failed rabit worker. missing these will break recovery.

Reply:
new as stored in checkpoint? all of those.
new as part of configuration? none.
checkpoint load overwrite configuration read so easiest way to fix this is to make checkpoint self contained.